### PR TITLE
[Core] Remove job driver logger

### LINKED ISSUE #47389: Core: File descriptor of ray job does not get closed
### What happened + What you expected to happen

File descriptor does not get closed in the job manager process on this line when long running job cluster keep receiving job. https://github.com/ray-project/ray/blob/e164d87a49f8b5c7bc5778859873d75845705b47/python/ray/dashboard/modules/job/job_manager.py#L91 leading to high cpu usage 

[
Screenshot 2024-08-28 at 8 08 38 PM
](url)

### Versions / Dependencies

Ray 2.34
Python 3.11

### Reproduction script

Ray jobs on a cluster with some fixed interval (e.g: 30 seconds)

### Issue Severity

High: It blocks me from completing my task.

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

This looks like leaking fd without writing any useful info

## Related issue number

Closes https://github.com/ray-project/ray/issues/47389

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by rkooo567** [comment]
probably it is better keeping the log but just closing the fd immediaetly

File: python/ray/dashboard/modules/job/job_manager.py [hunk 1]
```diff
@@ -84,23 +84,6 @@ def __init__(self, gcs_aio_client: GcsAioClient, logs_dir: str):
         self._recover_running_jobs_event = asyncio.Event()
         run_background_task(self._recover_running_jobs())
 
-    def _get_job_driver_logger(self, job_id: str) -> logging.Logger:
-        """Return job driver logger to log messages to the job driver log file.
-
-        If this function is called for the first time, configure the logger.
-        """
-        job_driver_logger = logging.getLogger(f"{__name__}.driver-{job_id}")
```
Comment (GeneDer) [inline python/ray/dashboard/modules/job/job_manager.py:92]: I kinda feel the right solution is maybe just have a fixed logger to log it like
```
job_driver_logger = logging.getLogger("ray.job.driver")
```

Comment (GeneDer) [reply python/ray/dashboard/modules/job/job_manager.py:92]: (and some kind of dummy path to log those messages)

We can update the message to include the submission id

Comment (rkooo567) [reply python/ray/dashboard/modules/job/job_manager.py:92]: the issue is that this logger wants to write to a file, and that file is never closed once it is open. So although we use 1 logger, we should keep changing files to write. 

I will just keep this as is, but just close the fd as soon as we write this line of log

Comment (GeneDer) [reply python/ray/dashboard/modules/job/job_manager.py:92]: I'm just worried if in the future, we are using this logger to log some other info, we might find the file is already closed and this logger being unusable. But if you think close the fd after each log statement is easier to implement, then I'm also fine with it :)

Comment (rkooo567) [reply python/ray/dashboard/modules/job/job_manager.py:92]: let me think about it a little more about the good solution.

**Pull request closed without merging**